pref-adjust.lua

```diff
@@ -91,16 +136,201 @@ function clearpref_all_dwarves()
     end
 end
 -- ---------------------------------------------------------------------------
-function adjust_all_dwarves()
+function adjust_all_dwarves(profile)
     for _,v in ipairs(df.global.world.units.active) do
         if v.race == df.global.ui.race_id then
             print("Adjusting "..dfhack.TranslateName(dfhack.units.getVisibleName(v)))
-            brainwash_unit(v)
+            brainwash_unit(v,profile)
         end
     end
 end
+-- ---------------------------------------------------------------------------
+function build_all_lists(printflag)
+    list_of_inorganics={} -- Type 0 "Likes iron.."
+    list_of_inorganics_string=""
+    vec=df.global.world.raws.inorganics -- also df.global.world.raws.inorganics_subset[0].id available
+    for k=0,#vec-1 do
+        name=vec[k].id
+        list_of_inorganics[name]=k
+        list_of_inorganics_string=list_of_inorganics_string..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 0 INORGANICS:"..list_of_inorganics_string) --    printall(list_of_inorganics)
+    end
+-- ------------------------------------
+    list_of_creatures={} --dict[name]=number, Type 1/3 "Likes them for.." / "Detests .."
+    vec=df.global.world.raws.creatures.all
+    list_of_creatures_string=""
+    for k=0,#vec-1 do
+        name=vec[k].creature_id
+        list_of_creatures[name]=k
+        list_of_creatures_string=list_of_creatures_string..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 1,3 CREATURES:"..list_of_creatures_string) --    printall(list_of_creatures)
+    end
+-- ------------------------------------
+    list_of_plants={} -- Type 2 "Prefers to consume.." and Type 5 "Likes Plump Helmets for their rounded tops"
+    -- (TODO could have edible only, and/or a different list for each-all edible meat/plants/drinks)
+    list_of_plants_string=""
+    vec=df.global.world.raws.plants.all
+    for k=0,#vec-1 do
+        name=vec[k].id
+        list_of_plants[name]=k
+        list_of_plants_string=list_of_plants_string..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 2,5 PLANTS:"..list_of_plants_string) --    printall(list_of_plants)
+    end
+-- ------------------------------------
+    list_of_items={} -- Type 4 "Likes armor.." (TODO need recursive material decode lists?)
+    list_of_items_string=""
+    -- [lua]# @dfhack.matinfo.decode(31,1)
+    -- <material 31:1 CREATURE:TOAD_MAN:GUT>
+    -- [lua]# @dfhack.matinfo.decode(0,1)
+    -- <material 0:1 INORGANIC:GOLD>
+    for k,v in ipairs(df.item_type) do
+        list_of_items[v]=k
```

Comment:
Side note: you didn't need `list_of_items` in the first place. `df.item_type` and any other DF enum can be indexed by both key and value. Example:
```
[lua]# ~df.item_type[0]
BAR
[lua]# ~df.item_type.BAR
0
```

Reply:
For consistency with the others, which don't correspond to DF enums, I think this is fine.